include/algorithms/synthesis/dd_synthesis.hpp

```diff
@@ -28,9 +29,21 @@ namespace syrec {
         }
 
     private:
-        double                 runtime  = 0.;
-        std::size_t            numGates = 0U;
-        qc::QuantumComputation qc;
+        double                                  runtime  = 0.;
+        std::size_t                             numGates = 0U;
+        std::unique_ptr<dd::Package<>>          ddSynth;
```

Comment:
Since you opted for storing the package as a member and still allow to pass a package instance to some functions, it would probably make sense to turn this (and the respective function parameters) into a shared pointer and set the member variable for the functions that receive a package from the outside.

And maybe just call the member `dd`

Reply:
The `getDD` used in `applyOperation` requires std::unique_ptr. If I make `dd` a shared pointer, how do I pass this to the getDD operation?

Reply:
Hm. Good point. Doesn't really make sense there as well. All of these functions should take a reference to a dd package. The package must not be null anyways.

It can't really be unique because that would assume you'd have to take ownership of the DD package that is passed to the function. You would lose the absolute to use the packe outside of the synthesizer.

Reply:
so what do you suggest we do? the getDD is from dd package right?

Reply:
It's from the QFR. The proper solution would be to create a PR there and change the respective function signatures to package references.
If you are up for that, then go ahead. If not, then probably just leave this as a unique pointer here and try to work with that.

Reply:
I don't think I can go around without using `getDD`. 

So what I propose for now: Leave everything as it is (I still need to incorporate the one pass synthesis solution as well and want to finish this early). Once all the methods (which I'm planning to implement) are incorporated, I will create a PR by forking QFR and then make the necessary changes in the dd synthesis files